app/src/common/utils/validation/validate.js

```diff
@@ -54,11 +54,16 @@ export const projectName = composeValidators([isNotEmpty, regex(/^[0-9a-zA-Z-_]{
 export const btsIntegrationName = composeValidators([isNotEmpty, maxLength(55)]);
 export const btsProject = composeValidators([isNotEmpty, maxLength(55)]);
 export const patternNameLength = composeValidators([isNotEmpty, maxLength(55)]);
+export const ruleNameLength = composeValidators([isNotEmpty, maxLength(55)]);
 export const createPatternNameUniqueValidator = (patternId, patterns) => (newPatternName) =>
   !patterns.some(
     ({ id, name: patternName }) =>
       patternName.toLowerCase() === newPatternName.toLowerCase() && id !== patternId,
   );
+export const createRuleNameUniqueValidator = (notifications) => (newRuleName) =>
+  !notifications.some(
+    ({ ruleName }) => ruleName && ruleName.toLowerCase() === newRuleName.toLowerCase(),
```

Comment:
You need to check notification id too as for patterns. This helps to avoid false-negative validation when editing the notification.

Reply:
And form where should I get this id ?

Reply:
Yes, it will be added to the notification rule model.